test/prometheus-manifests/service-monitor-kubelet.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```

Comment:
Could you make this ServiceMonitor more minimal? We shouldn't need all the relabeling rules here and the only path that we want to scrape on Kubelet is `/metrics/resources`.

Reply:
I made the ServiceMonitor more minimal. However, I'm still scraping `/metrics/cadvisor`, not `/metrics/resource`, which do not provide `container_cpu_usage_seconds_total{id='/'}`.

Reply:
It was renamed to `container_cpu_usage_seconds` in https://github.com/kubernetes/kubernetes/pull/86282. Since `/metrics/resource` is a more lightweight version of `/metrics/cadvisor`, it would be better to use the new endpoint and the new metric.

Reply:
Actually, https://github.com/kubernetes/kubernetes/pull/86282 was kind of reverted by https://github.com/kubernetes/kubernetes/pull/89540 in Kubernetes 1.19, so the metric is named `container_cpu_usage_seconds_total` (for k8s < 1.18 and k8s >= 1.19), and that's the metric [documented](https://kubernetes.io/docs/reference/instrumentation/metrics/).

What I meant is that, using the path `/metrics/resource`, query`container_cpu_usage_seconds_total{id='/'}` returns nothing, because the label `id` is never empty (however it works on path `/metrics/cadvisor`).

So I've updated the configmap manifest to use `node_cpu_usage_seconds_total` and `node_memory_working_set_bytes` instead. I can make a separate PR for that if you prefer. I was unsure because you suggested yourself in https://github.com/kubernetes-sigs/prometheus-adapter/pull/531#discussion_r961844262 to use `{id='/'}` for node queries.

Reply:
> What I meant is that, using the path /metrics/resource, querycontainer_cpu_usage_seconds_total{id='/'} returns nothing, because the label id is never empty (however it works on path /metrics/cadvisor

Oh that's interesting, I wasn't aware that `id` wasn't included in the metric exposed `/metrics/resource`. I guess an alternative could then be to group by `node` label since it is injected by prometheus-operator, but at this point I am not really sure what's best between that and depending on node-exporter. For sure we should move away from /metrics/cadvisor since it is bound to disappear.